assets/src/components/form/base/auto-form-row.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External imports
+ */
+import { Component } from 'react';
+import { isArray } from 'lodash';
+
+/**
+ * Internal imports
+ */
+import { mapValidatorsToAttributes } from '../inputs/base/utils';
+
+/**
+ * AutoFormRow
+ * This component determines whether the incoming form element
+ * is an input, label, or row component and responds accordingly.
+ * FormRow components from a form layout strategy will simply be
+ * returned as is, since they require no further manipulation.
+ * Inputs and Labels will be passed to the supplied form layout
+ * strategy AutoColumnRow component for further processing.
+ * If a Label component is not provided but the supplied Input
+ * has a "label" prop, then that will be used to generate an
+ * InputLabel using the supplied form layout strategy component.
+ *
+ * @component
+ * @param {Object} FormElement either an input, label, or form row component
+ * @param {Object} InputLabel form layout strategy label display component
+ * @param {Object} AutoColumnRow form layout strategy row component
+ * @return {Object} rendered form row
+ */
+export class AutoFormRow extends Component {
+	render() {
+		const {
+			FormElement,
+			InputLabel,
+			AutoColumnRow,
+		} = this.props;
+
+		let label = null;
+		let input = null;
+		let row = null;
+		if ( FormElement.type ) {
+			const name = FormElement.type.name;
```

Comment:
Those incoming props are the unrendered react components, and type is indeed an object. 
If the name had simply been the value of `FormElement.type` then none of the code within this conditional would have worked.

Reply:
> If the name had simply been the value of FormElement.type then none of the code within this conditional would have worked.

Since there are no tests and there's no clear indication that the code is being transpiled to anything in this branch, I wasn't certain this was verified to be "working" yet, hence my question :)